src/main/java/net/coderbot/iris/Iris.java

```diff
@@ -91,6 +100,21 @@ private void loadExternalShaderpack(String name) {
 		logger.info("Using shaderpack: " + name);
 	}
 
+	private static Optional<Path> loadExternalZipShaderpack(Path shaderpackPath) {
+		try {
+			FileSystem zipSystem = FileSystems.newFileSystem(shaderpackPath, Iris.class.getClassLoader());
```

Comment:
It's possible that this could cause a FileSystemAlreadyExistsException since you never close the created filesystem. This will cause issues with shaderpack reloading. It would be good to store this created filesystem so that it can be closed later. (such as when the shaderpack is reloaded)

Reply:
This won't happen by itself but the reload PR could cause issues.

Reply:
Okay, ill just store it for now, and if this pr gets merged i'll fix it in the reloading pr.